docs/developers/quickstart/deploy-smart-contract/truffle.md

```diff
@@ -0,0 +1,194 @@
+
+To create an empty Truffle project, run:
+
+```bash
+truffle init zkevm-tutorial
+```
+
+Change into the new directory:
+
+```bash
+cd zkevm-tutorial
+```
+
+## Write the smart contract
+
+Create your smart contract in the `contracts` directory by either creating a new file `Token.sol` or calling `truffle create contract Token`. Then, add the following code:
+
+```sol
+pragma solidity 0.8.17;
+
+// SPDX-License-Identifier: MIT
+
+contract Token {
+  string public name = "My Token";
+  string public symbol = "MTK";
+  uint8 public decimals = 18;
+  uint256 public totalSupply = 100000000;
+
+  mapping (address => uint256) public balances;
+  address public owner;
+
+  constructor() {
+    owner = msg.sender;
+    balances[owner] = totalSupply;
+  }
+
+  function transfer(address recipient, uint256 amount) public {
+    require(balances[msg.sender] >= amount, "Insufficient balance.");
+    balances[msg.sender] -= amount;
+    balances[recipient] += amount;
+  }
+}
+```
+
+:::warning Do not use this contract code in production
+
+The above contract is for testing purposes and has not been audited.
+
+:::
+
+You can check if it compiles by running `truffle compile` from the root folder.
+
+## Write the migration script
+
+To tell Truffle how, and in what order we want to deploy our smart contracts, we need to write a migration script.
+
+Create `1_deploy_token.js` in the `migrations` directory, and add the following code:
+
+```javascript
+const Token = artifacts.require("Token");
+
+module.exports = function (deployer) {
+  deployer.deploy(Token);
+};
+```
+
+## Deploy your contract
+
+Truffle offers two ways of deploying your contracts, through the [Truffle Dashboard](#truffle-dashboard) or through the [CLI by modifying `truffle-config.js`](#truffle-configjs)
+
+### Truffle Dashboard
+
+[Truffle Dashboard](https://trufflesuite.com/docs/truffle/how-to/use-the-truffle-dashboard/) allows you to forgo saving your private keys locally, instead connecting to your MetaMask wallet for deployments. To deploy with Truffle Dashboard, you need to:
+
+1. Run `truffle dashboard` in your terminal, which will open a window on port `24012`.
+1. Navigate to `localhost:24012` in your browser. Please ensure that Dashboard is connected to the ConsenSys zkEVM testnet by connecting your MetaMask wallet to ConsenSys zkEVM. For reference, the ConsenSys zkEVM testnet network ID is `59140`.
+
+   ![confirm network](../../../assets/dashboard_network.png)
+
+1. Run `truffle migrate --network dashboard` in a separate terminal.
+1. Navigate back to `localhost:24012`. You should see a prompt asking your to confirm the deployment. Click **Confirm**.
+
+   ![confirm deployment](../../../assets/dashboard_deploy.png)
+
+### truffle-config.js
+
+You can deploy with Truffle using the command line, by specifying the ConsenSys zkEVM network in `truffle-config.js`. To do so, you need to:
+
+1. Create a `.env` file in the `truffle` folder with your wallet's mnemonic and Infura API key.
```

Comment:
Same as with the Hardhat instructions. The example here uses the private key (not mnemonic)? Also, shouldn't we create the `.env` in the root directory of the project?

Reply:
truffle uses the 12 word mnemonic HH uses the private key

Reply:
but yeah I'll change the private key to. menmonic